### DISCUSSION TITLE
Integrate the filepond image validate size plugin

### DISCUSSION THREAD

**Discussion by yob-yob** [filamentphp/filament#5037] [Ideas]
# Integrate the filepond image validate size plugin

It would be a nice feature that Filepond integrates the image validate size plugin as this allows us validate images on the frontend rather than validating them on the backend...

https://pqina.nl/filepond/docs/api/plugins/image-validate-size/

the main concern that I have is that filepond shows a green signal (successful) when the file is uploaded even if LATER ON when the form is submitted, it will return an error when the `dimension` validation will fail...

To temporarily fix this I setup a realtime validation for file-uploading

```
$filepond->reactive()
                ->afterStateUpdated(function ($state) {
                  $validator = Validator::make([
                    'photo' => $state
                  ], [
                    'photo' => [
                        'mimes:jpg,png',    
                        'dimensions:min_width=1200,min_height=1200,max_width=5000,max_height=5000'
                    ]
                  ]);
                  $validator->validate();
                })
```

but with this temporary fix, the Green Signal will still be visible...

There is however an `imageResizeTargetWidth` the problem I have with this is that I don't need to have a fixed image size requirements... I want to user to have some freedom on the images they upload but with some limitations as well.

P.S. I would also suggest that filament will make `imageResizeUpscale` to be configurable...

**Comment by adevade** [comment]
I was looking for this functionality as well. 

Being able to set the allowed `min` and `max` `width` and `height` to be uploaded would be really useful. We need to stop clients from uploading small thumbnails and using them as hero images&hellip; 😅

**Comment by binaryfire** [comment]
This would be a great addition, especially in combination with @saade’s latest PR: https://github.com/filamentphp/filament/pull/9394

You need as much frontend validation as possible when bypassing the server and uploading directly to S3.

This discussion has been open for a year now so I’m not sure if the core team have seen it. If my JS skills were better I’d create a PR… I’m guessing it’d be pretty easy to add though.
